### PR TITLE
Test that Wrapping arithmetic ops are implemented for all int types

### LINKED ISSUE #49660: Add tests for ensuring we have the right impls for arithmetic traits
The other day I noticed that `Wrapping` only implements `Shl` and co. for `usize`, which contradicts that without `Wrapping` the shift amount can be any type. So, I think it may be useful to create tests which use macros to shove together all the types we expect to have impls for and check that we're consistent there.

What I've noticed:

* For every operation three extra impls are added, with the combinations of single-level references on both sides, e.g. `&T + T`, `T + &T`, `&T + &T` for `Add`.
* Assignment ops do the same as the above, with only one extra impl, e.g. `T += &T` for `Add`.
* `Add`, `Sub`, `Mul`, `Div`, `Rem` exist for all int and float types with themselves
* `BitAnd`, `BitOr`, `BitXor` exist for all int types and bool
* `Shl` and `Shr` allow any integer type for both the left and the right side
* `Neg` exists for all ints and floats
* `Not` exists for all ints and bool
* `Wrapping` should mimic all of the above, where references are of the form `&Wrapping<T>` and not `Wrapping<&T>`.

Like, I wouldn't be surprised if rearranging some macros might add or remove impls for these and adding comprehensive tests might help make sure we covered everything.

### PR DESCRIPTION
**Pull request by iluuu1994**
Closes #49660

### CODE REVIEW

File: src/libcore/num/wrapping.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ macro_rules! sh_impl_signed {
                 }
             }
         }
+        forward_ref_binop! { impl Shl, shl for Wrapping<$t>, $f }
```
Comment (iluuu1994) [inline src/libcore/num/wrapping.rs]: Oh, we probably need a `#[stable(since = ...]` annotation, right?

Comment (sfackler) [reply src/libcore/num/wrapping.rs]: Yeah, you may need to modify the macro definition to allow for that.

Comment (iluuu1994) [reply src/libcore/num/wrapping.rs]: Ok 👍 What's the Rust version that will release this? 1.38.0?

Comment (sfackler) [reply src/libcore/num/wrapping.rs]: 1.39 I believe.

Comment (iluuu1994) [reply src/libcore/num/wrapping.rs]: @sfackler What feature name should I use?

Comment (sfackler) [reply src/libcore/num/wrapping.rs]: It doesn't matter - `wrapping_ref_ops` or whatever should be fine.

**Comment by sfackler** [comment]
Since trait impls are immediately stable, let's go through FCP, but this seems pretty uncontroversial since it's just adding the by-reference impls that primitives already have.

@rfcbot fcp merge

**Comment by sfackler** [comment]
Yep - r=me with tests for usize/isize.

**Comment by iluuu1994** [comment]
Can this test be skipped for asm.js?

**Comment by sfackler** [comment]
You can probably `#[cfg]` off the 128 bit tests on emscripten, yeah.

**Comment by iluuu1994** [comment]
Should be good now.

**Comment by iluuu1994** [comment]
Can we re-run the tests?

**Comment by iluuu1994** [comment]
Thanks @sfackler!

**Pull request merged**